processing/src/main/java/io/druid/query/AbstractQueryMetrics.java

```diff
@@ -0,0 +1,203 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Strings;
+import com.google.common.collect.Iterables;
+import com.metamx.emitter.service.ServiceEmitter;
+import com.metamx.emitter.service.ServiceMetricEvent;
+import org.joda.time.Interval;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+public abstract class AbstractQueryMetrics<QueryType extends Query<?>> implements QueryMetrics<QueryType>
```

Comment:
I'm against fragmentation of `QueryMetrics` implementations. The other day some other query type added or changed that will require some of existing `QueryMetrics` methods, and neither `AbstractQueryMetrics` nor `DefaultQueryMetrics` will help.

I suggest to remove `dataSource()`, `interval()` and `duration()` methods from `QueryMetrics` and instead add a single method `dataSourcesAndIntervalsAndDurations()`.